### PR TITLE
prometheus: added options for alertmanager

### PR DESCRIPTION
**Pull request by oida**
###### Motivation for this change

Additional options for Prometheus so it can be connected with Prometheus Alertmanager.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/default.nix [hunk 1]
```diff
@@ -388,6 +391,30 @@ in {
           A list of scrape configurations.
         '';
       };
+
+      alertmanagerURL = mkOption {
+        type = types.str;
```
Comment (groxxda) [inline nixos/modules/services/monitoring/prometheus/default.nix]:
```
type = types.listOf types.str;
default = [];
description = ''
  List of ...
''
```

File: nixos/modules/services/monitoring/prometheus/default.nix [hunk 2]
```diff
@@ -29,6 +29,9 @@ let
     "-storage.local.path=${cfg.dataDir}/metrics"
     "-config.file=${writePrettyJSON "prometheus.yml" promConfig}"
     "-web.listen-address=${cfg.listenAddress}"
+    "-alertmanager.notification-queue-capacity=${toString cfg.alertmanagerNotificationQueueCapacity}"
+    "-alertmanager.timeout=${toString cfg.alertmanagerTimeout}s"
+    (optionalString (cfg.alertmanagerURL != "") "-alertmanager.url=${cfg.alertmanagerURL}")
```
Comment (groxxda) [inline nixos/modules/services/monitoring/prometheus/default.nix]: `cfg.alertmanagerURL != []`

**Review by groxxda — CHANGES_REQUESTED** [review]
Upstreamen nicht vergessen ^^

**Review by groxxda — APPROVED** [review]

**Review by globin — APPROVED** [review]
LGTM

**Pull request merged**